src/ansiblelint/rules/yaml.py

```diff
@@ -54,6 +54,7 @@ def matchyaml(self, file: Lintable) -> List["MatchError"]:
         for problem in run_yamllint(
             file.content, YamllintRule.config, filepath=file.path
         ):
+            self.description = problem.desc
```

Comment:
At first glance this keeps overwriting description in a loop, so only the last one wins. It is not obvious that this relies on a side effect of `create_matcherror()` which does `rule=copy.copy(self)`.

So, I would rather do something like:

```python
match = self.create_matcherror(
    ...
)
// explain why we're doing this here
match.rule.description = problem.desc
matches.append(match)
```

Reply:
@duncanmmacleod Unless you address this issue, this fix might fail to be included in 6.2.0 release. Usually I would have tried to fix it myself but I am at 110% already with that release.

Reply:
Sorry @ssbarnea, I was waiting to see the outcome of the discussion on #2106 before going much further. I'm not desparate to see this deployed immediately, so if I miss this release that's not the end of the world for me.